crates/npmrc/src/custom_deserializer.rs

```diff
@@ -22,19 +22,21 @@ pub fn default_public_hoist_pattern() -> Vec<String> {
 /// On macOS: ~/Library/pacquet/store
 /// On Linux: ~/.local/share/pacquet/store
 pub fn default_store_dir() -> PathBuf {
-    // TODO: If env variables start with ~, make sure to resolve it into home_dir.
+    let home_dir = dirs::home_dir().expect("Home directory is not available");
+
     if let Ok(pacquet_home) = env::var("PACQUET_HOME") {
-        return PathBuf::from(pacquet_home).join("store");
+        // if variable start with `~/`,it should be removed,or use default value.
+        let pacquet_home_without_tilde =
+            pacquet_home.strip_prefix("~/").unwrap_or(pacquet_home.as_str());
```

Comment:
The name `pacquet_home_without_tilde` is too long, the name `pacquet_home` isn't used after this line. You can rename `pacquet_home_without_tilde` to `pacquet_home` (Rust supports [variable shadowing](https://doc.rust-lang.org/rust-by-example/variable_bindings/scope.html)).